server/router/src/main/java/org/infinispan/server/router/router/impl/hotrod/HotRodRouter.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.server.router.router.impl.hotrod;
+
+import java.lang.invoke.MethodHandles;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Optional;
+
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.server.router.RoutingTable;
+import org.infinispan.server.router.configuration.HotRodRouterConfiguration;
+import org.infinispan.server.router.logging.RouterLogger;
+import org.infinispan.server.router.router.Router;
+import org.infinispan.server.router.router.impl.hotrod.handlers.SniHandlerInitializer;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.ServerSocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+/**
+ * {@link Router} implementation for Hot Rod.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class HotRodRouter implements Router {
+
+    private static final RouterLogger logger = LogFactory.getLog(MethodHandles.lookup().lookupClass(), RouterLogger.class);
+
+    private static final String THREAD_NAME_PREFIX = "MultiTenantRouter";
+
+    private final NioEventLoopGroup masterGroup = new NioEventLoopGroup(1, new DefaultThreadFactory(THREAD_NAME_PREFIX + "ServerMaster"));
+    private final NioEventLoopGroup workerGroup = new NioEventLoopGroup(0, new DefaultThreadFactory(THREAD_NAME_PREFIX + "ServerWorker"));
```

Comment:
Then I think you should move the `NioEventLoopGroup` initialization to the `start()` method.